lib/keycard/client_resolver.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Keycard
+  # This resolver should be used when the application will serve HTTP requests
+  # directly or through a proxy that sets the usual/trusted HTTP headers.
+  class ClientResolver
+    def attributes_for(request)
+      {
+        username: request.env['REMOTE_USER'],
+        client_ip: request.env['REMOTE_ADDR']
```

Comment:
Looking at Rack::Request it looks like in at least some cases `REMOTE_ADDR` can be multi-valued. Again, I'm wondering how the application would use this info. In particular, if it's to pass it through for logging and the expectation is that there's a single IP address here, I think we might need some more logic here.